tests/functional/test_resolve_packse.py

```diff
@@ -0,0 +1,166 @@
+)
+
+EXPECTED_TO_FAIL = {
+    "example": (
+        "Expected solution looks wrong: "
+        "https://github.com/astral-sh/packse/issues/157"
+    ),
+    "local-not-used-with-sdist": (
+        "Tests that sdist versions are preferred over local versions. "
+        "Discussed: https://github.com/astral-sh/packse/issues/158. "
+        "TODO: Find if this is a known issue on pip and/or packaging"
+    ),
+    "local-transitive-confounding": (
+        "Expected solution looks wrong: "
+        "https://github.com/astral-sh/packse/issues/159"
+    ),
+    "transitive-prerelease-and-stable-dependency": (ambigious_prerelease_resolution),
+    "transitive-prerelease-and-stable-dependency-many-versions": (
+        ambigious_prerelease_resolution
+    ),
+    "transitive-package-only-prereleases-in-range-opt-in": (
+        ambigious_prerelease_resolution
+    ),
+    "package-only-prereleases-boundary": (
+        "Expected solution is probably wrong: "
+        "https://github.com/astral-sh/packse/issues/161"
+    ),
+    "package-prereleases-specifier-boundary": (requires_different_python_version),
+    "python-greater-than-current": (requires_different_python_version),
+    "python-greater-than-current-patch": (requires_different_python_version),
+    "python-greater-than-current-backtrack": (requires_different_python_version),
+    "python-greater-than-current-excluded": (requires_different_python_version),
+    "compatible-python-incompatible-override": (requires_different_python_version),
+    "incompatible-python-compatible-override-unavailable-no-wheels": (
+        requires_different_python_version
+    ),
+    "incompatible-python-compatible-override-no-compatible-wheels": (
+        requires_different_python_version
+    ),
+    "incompatible-python-compatible-override-other-wheel": (
+        requires_different_python_version
+    ),
+    "python-patch-override-no-patch": (requires_different_python_version),
+    "package-only-yanked": (not_served_as_yanked),
+    "package-only-yanked-in-range": (not_served_as_yanked),
+    "requires-package-yanked-and-unyanked-any": (not_served_as_yanked),
+    "package-yanked-specified-mixed-available": (not_served_as_yanked),
+    "transitive-package-only-yanked": (not_served_as_yanked),
+    "transitive-package-only-yanked-in-range": (not_served_as_yanked),
+    "transitive-yanked-and-unyanked-dependency": (not_served_as_yanked),
+}
+
+
+def run_command(command: List[str], cwd: None = None) -> str:
+    result = subprocess.run(command, cwd=cwd, capture_output=True, text=True)
+    result.check_returncode()
+    return result.stdout.strip()
+
+
+@pytest.fixture(scope="session", autouse=True)
+def start_packse_server() -> Generator[None, None, None]:
+    """Starts the packse server before tests run and ensures it's terminated after."""
+    proc = subprocess.Popen(
+        ["packse", "serve", "--host", "127.0.0.1", "--port", "3141"]
+    )
+    time.sleep(1)
+    yield
+    proc.terminate()
```

Comment:
I would recommend using the static index packse generates at https://github.com/astral-sh/packse/issues/165#issuecomment-2002510480 instead — the `serve` command is intended for development purposes.

Reply:
My concern with that, and maybe it's not justified, is that tests could fail because the scenarios from `packse fetch` & `packse inspect` would not align perfectly with the scenarios on the index.

Reply:
`fetch` defaults to pulling the current packse version tag which should always match the corresponding index for that tag. You just need to do `packse inspect --no-hash` since the index is versioned we don't need to include a hash to uniquely identify scenarios.

Reply:
I had a long throught about this and my main concern is that if pip wants to add it's own scenarios, or deviate away from packse scenarios (e.g. because they seemingly don't follow the spec https://github.com/astral-sh/packse/issues/161) then using the astra index won't work.

However, for now, I will attempt to implement using the index, and that problem can be worried about in the future.